scripts/ceph-windows/collect-event-logs.ps1

```diff
@@ -0,0 +1,45 @@
+param (
+    [Parameter(Mandatory)]
+    [string]$LogDirectory,
+    [switch]$IncludeEvtxFiles = $false,
+    [switch]$CleanupEventLog = $false
+)
+
+function DumpEventLogEvtx($path){
+    foreach ($i in (Get-WinEvent -ListLog * |  ? {$_.RecordCount -gt 0 })) {
+        $logName = "eventlog_" + $i.LogName + ".evtx"
+        $logName = $logName.replace(" ","-").replace("/", "-").replace("\", "-")
+        Write-Host "exporting "$i.LogName" as "$logName
+        $logFile = Join-Path $path $logName
+        & $Env:WinDir\System32\wevtutil.exe epl $i.LogName $logFile
```

Comment:
In PowerShell, any `.exe` binary executed from the disk needs to be explicitly checked for `$LASTEXITCODE`.
Something like:
```powershell
if($LASTEXITCODE) {
    Throw "Failed to export $($i.LogName) to $logFile"
}
```

Only native PowerShell functions / cmdlets generate exceptions on error, and don't need any extra error checking.